ScanbotSDKExamples/NET/ReadyToUseUI.Droid/MainActivity.cs

```diff
@@ -294,155 +405,199 @@ protected override void OnActivityResult(int requestCode, [GeneratedEnum] Result
                 return;
             }
 
-            if (requestCode == Constants.CAMERA_DEFAULT_UI_REQUEST_CODE)
+            if (!scanbotSDK.LicenseInfo.IsValid)
             {
-                var parcelable = data.GetParcelableArrayExtra(RtuConstants.ExtraKeyRtuResult);
-                var pages = parcelable.Cast<Page>().ToList();
-
-                PageRepository.Add(pages);
-                var intent = new Intent(this, typeof(PagePreviewActivity));
-                StartActivity(intent);
+                Alert.ShowLicenseDialog(this);
+                return;
             }
-            else if (requestCode == Constants.IMPORT_IMAGE_REQUEST)
+
+            switch (requestCode)
             {
-                if (!SBSDK.IsLicenseValid())
+                case SCAN_DOCUMENT_REQUEST_CODE:
+                case SCAN_DOCUMENT_WITH_FINDER_REQUEST_CODE:
                 {
-                    Alert.ShowLicenseDialog(this);
+                    var parcelable = data.GetParcelableArrayExtra(RtuConstants.ExtraKeyRtuResult);
+
+                    StartActivity(new Intent(this, typeof(PagePreviewActivity)));
                     return;
                 }
-
-                progress.Visibility = ViewStates.Visible;
-                Alert.Toast(this, Texts.importing_and_processing);
-                Task.Run(delegate
+                case IMPORT_IMAGE_REQUEST:
                 {
-                    var result = Utils.ImageUtils.ProcessGalleryResult(this, data);
+                    progress.Visibility = ViewStates.Visible;
+
+                    Alert.Toast(this, Texts.importing_and_processing);
 
-                    var pageId = SBSDK.PageStorage.Add(result);
+                    var result = ImageUtils.ProcessGalleryResult(this, data);
+
+                    var pageId = pageStorage.Add(result);
                     var page = new Page(pageId, new List<PointF>(), DetectionStatus.Ok, ImageFilterType.None);
-                    page = SBSDK.PageProcessor.DetectDocument(page);
-                    PageRepository.Add(page);
+                    page = scanbotSDK.CreatePageProcessor().DetectDocument(page);
 
-                    var intent = new Intent(this, typeof(PagePreviewActivity));
-                    RunOnUiThread(delegate
-                    {
-                        progress.Visibility = ViewStates.Gone;
-                        StartActivity(intent);
-                    });
-                });
-            }
-            else if (requestCode == Constants.IMPORT_BARCODE_REQUEST)
-            {
-                Task.Run(delegate
+                    progress.Visibility = ViewStates.Gone;
+
+                    StartActivity(new Intent(this, typeof(PagePreviewActivity)));
+                    return;
+                }
+                case QR_BARCODE_DEFAULT_REQUEST:
```

Comment:
@yurii-scanbot, while sharing the same implementation for each case is a good idea, it creates a challenge for the documentation snippet because all three cases will be visible even when only focusing on one type of barcode scanner. 

Since the implementation is the same and the code is only used in the context of our app, it makes sense to simply have just one (QR_BARCODE_DEFAULT_REQUEST or maybe even BARCODE_DEFAULT_REQUEST).